### ISSUE TITLE
Do we really want one ZIM per curriculum and per language?

### ISSUE THREAD

**Issue by benoit74** [openzim/freecodecamp#45]
This issue is more here to bring awareness on an important ZIM "design" question which has not been discussed publicly AFAIK.

Currently, the scraper is capable to create one ZIM per list of courses and soon per language. 

And basically, we are currently planning to create one ZIM per curriculum (set of courses as exposed on https://www.freecodecamp.org/learn/ and per language.

We could wonder if we shouldn't make less ZIMs. I don't have certainties, but at least I wanna note down my arguments and bring awareness to this important decision.

## Group all curriculum

While it is neat to have many ZIMs in the library (looks like you've done lot of worker and have lot of things), it make the life of content selectors (persons who build SD cards for hotspots, ...) harder. And it does not help final users to discover all curriculum available or easily switch from one to the next.

Grouping all curriculum in one ZIM is possible and would even provide a UX similar to what is online.

The drawback is that if you're only interested in javascript curriculum (which is very small) you will still have to pay the price (in terms of DL and storage) for big videos of some other curriculum (e.g. Python).

## Have all language at hand

One could argue this is the same for languages. And that switching language easily in the UI could help someone to learn English (try to do as much of the course as possible in English, and switch to native language when you don't understand a sentence).

I'm not really convinced by the idea

## Conclusion

No real conclusion, again this issue is more here to bring awareness on this design decision

**Comment by benoit74** [comment]
@kelson42 @Popolechien Any views ? (no need to have strong PoV on this one, again this is just to bring awareness and not realize too late that we do not have what we would like to have)

**Comment by kelson42** [comment]
Yes, I confirm that one ZIM per language and curriculum still seems to me the best approach.

**Comment by benoit74** [comment]
Thank you for confirming this! 

Regarding spoken languages, the whole curriculum are translated in 11 languages, meaning we have 11 sets of curriculum files, and the scraper selects the appropriate one based on setting, so no risk to mix them / miss-label them.
